clients/src/main/java/org/apache/kafka/clients/LeastLoadedNode.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+import org.apache.kafka.common.Node;
+
+public class LeastLoadedNode {
+    private final Node node;
+    private final boolean atLeastOneConnected;
+
+    public LeastLoadedNode(Node node, boolean atLeastOneConnected) {
+        this.node = node;
+        this.atLeastOneConnected = atLeastOneConnected;
+    }
+
+    public Node node() {
+        return node;
+    }
+
+    public boolean isAtLeastOneConnected() {
```

Comment:
We are setting this to true only for ready connections, should we rename the method to indicate that?

Reply:
Done this, but https://github.com/apache/kafka/pull/13277#discussion_r1633579132 deprecates this anyway